{
    "cveId": "CVE-2021-29582",
    "version": "1.0.0",
    "timestamp": "2024-12-15T20:24:27.876763+00:00",
    "description": "TensorFlow is an end-to-end open source platform for machine learning. Due to lack of validation in `tf.raw_ops.Dequantize`, an attacker can trigger a read from outside of bounds of heap allocated data. The implementation(https//github.com/tensorflow/tensorflow/blob/26003593aa94b1742f34dc22ce88a1e17776a67d/tensorflow/core/kernels/dequantize_op.cc#L106-L131) accesses the `min_range` and `max_range` tensors in parallel but fails to check that they have the same shape. The fix will be included in TensorFlow 2.5.0. We will also cherrypick this commit on TensorFlow 2.4.2, TensorFlow 2.3.3, TensorFlow 2.2.3 and TensorFlow 2.1.4, as these are also affected and still in supported range.",
    "keyphrases": {
        "rootcause": "lack of validation in tf.raw_ops.Dequantize",
        "weakness": "read from outside of bounds of heap allocated data",
        "impact": "",
        "vector": "",
        "attacker": "attacker",
        "product": "TensorFlow",
        "version": "before 2.5.0",
        "component": "tf.raw_ops.Dequantize"
    }
}
